lib/models/types/boolean.ex

```diff
@@ -0,0 +1,12 @@
+defmodule SlackCoder.Models.Types.Boolean do
+  @behaviour Ecto.Type
+  def type(), do: :boolean
+  def cast(val), do: {:ok, value_to_boolean(val)}
+  def load(val), do: {:ok, value_to_boolean(val)}
+  def dump(val), do: {:ok, value_to_boolean(val)}
+
+  def value_to_boolean(on) when on in ["on", "yes", "1", "y"], do: true
+  def value_to_boolean(off) when off in ["off", "no", "0", "n"], do: false
+  def value_to_boolean(bool) when bool in [true, false], do: bool
+  def value_to_boolean(_), do: nil
+end
```

Comment:
@mtyeh411 This is a cool example of Ecto's custom types. Basically you define 4 functions `type` (the database primitive type), cast (input done prior to validation to put it into a validateable form), load (when it comes out of the DB how to load into memory) and finally dump (when we dump it to the DB, i.e. in memory version after cast to db primitive form).

This `Boolean` type accepts either real booleans `true`/`false` or strings like `"on"`, `"yes" or `"1"`. Elsewhere I make sure the input is lowercased, but for robustness you could do that here as well.